windows/syscall.go

```diff
@@ -51,6 +51,20 @@ func BytePtrFromString(s string) (*byte, error) {
 	return &a[0], nil
 }
 
+// UTF16PtrToString converts a *uint16 to a string
+func UTF16PtrToString(a *uint16) string {
+  if a == nil {
+    return ""
+  }
+  var i uintptr
+  ptr := uintptr(unsafe.Pointer(a))
+
+  // Iterate until we hit a null terminator
+  for i = 0; *(*uint16)(unsafe.Pointer(ptr + i)) != 0; i++ {
```

Comment:
We don't get the length, so we have to expect we'll get a null-terminated string.